library.lisp

```diff
@@ -78,6 +78,29 @@
                   #+(and unix (not darwin)) (env-paths "LD_LIBRARY_PATH")
                   #+darwin (env-paths "DYLD_LIBRARY_PATH"))))
 
+(defun elf-file-p (path)
+  (with-open-file (elf path :element-type '(unsigned-byte 8) :if-does-not-exist :error)
+    ;; All ELF files begin with the same four bytes
+    (and (= (read-byte elf) #x7f)
+         (= (read-byte elf) #x45)
+         (= (read-byte elf) #x4c)
+         (= (read-byte elf) #x46))))
+
+(defun follow-ld-script (path)
+  (if (elf-file-p path)
+      path
+      (let* ((file-lines (uiop:read-file-lines path))
+             (group-line (first (remove-if-not (lambda (line)
+                                                 (uiop:string-prefix-p "GROUP ( " line))
+                                               file-lines)))
+             (path-to-lib (fourth (uiop:split-string group-line :separator "( )"))))
+        path-to-lib)))
+
+(defun ensure-elf (path)
+  #+unix (follow-ld-script path)
```

Comment:
Please rename this function to `ensure-shared-library-file` and change the feature test to `#+linux`. Darwin is unix and does not use ELF or these delegate files.